addons/binding/org.openhab.binding.smartthings/src/main/java/org/openhab/binding/smartthings/handler/SmartthingsThingHandler.java

```diff
@@ -0,0 +1,331 @@
+     */
+    public SmartthingsThingHandler(Thing thing) {
+        super(thing);
+    }
+
+    /**
+     * Called when openHAB receives a command for this handler
+     *
+     * @param channelUID The channel the command was sent to
+     * @param command    The command sent
+     */
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        Bridge bridge = getBridge();
+
+        // Check if the bridge has not been initialized yet
+        if (bridge == null) {
+            logger.debug(
+                    "The bridge has not been initialized yet. Can not process command for channel {} with command {}.",
+                    channelUID.getAsString(), command.toFullString());
+            return;
+        }
+
+        SmartthingsBridgeHandler smartthingsBridgeHandler = (SmartthingsBridgeHandler) bridge.getHandler();
+        if (smartthingsBridgeHandler != null
+                && smartthingsBridgeHandler.getThing().getStatus().equals(ThingStatus.ONLINE)) {
+            String thingTypeId = thing.getThingTypeUID().getId();
+            String smartthingsType = getSmartthingsAttributeFromChannel(channelUID);
+
+            SmartthingsHttpClient httpClient = smartthingsBridgeHandler.getSmartthingsHttpClient();
+
+            SmartthingsConverter converter = converters.get(channelUID);
+            if (converter == null) {
+                logger.info("No converter available for channel {} with command {}. OpenHab can not update the device.",
+                        channelUID.getAsString(), command.toFullString());
+                return;
+            }
+
+            String path;
+            String jsonMsg;
+            if (command instanceof RefreshType) {
+                path = "/state";
+                // Go to ST hub and ask for current state
+                jsonMsg = String.format(
+                        "{\"capabilityKey\": \"%s\", \"deviceDisplayName\": \"%s\", \"capabilityAttribute\": \"%s\", \"openHabStartTime\": %d}",
+                        thingTypeId, smartthingsName, smartthingsType, System.currentTimeMillis());
+            } else {
+                // Send update to ST hub
+                path = "/update";
+                jsonMsg = converter.convertToSmartthings(channelUID, command);
+
+                // The smartthings hub won't (can't) return a response to this call. But, it will send a separate
+                // message back to the SmartthingBridgeHandler.receivedPushMessage handler
+            }
+
+            try {
+                httpClient.sendDeviceCommand(path, jsonMsg);
+                // Smartthings will not return a response to this message but will send it's response message
+                // which will get picked up by the SmartthingBridgeHandler.receivedPushMessage handler
+            } catch (InterruptedException | TimeoutException | ExecutionException e) {
+                logger.info("Attempt to send command to the Smartthings hub failed with: {}", e);
```

Comment:
Remove the last "{}". The coding guidelines require to add an exception as last argument, not part of the message. That way the logging framework not only calls "toString()" on the exception, but prints an actual stack trace.

If that is not what you want, you need to explicitly use exception.toString()